extra/utils/permission.js

```diff
@@ -0,0 +1,49 @@
+let permissions = {
+    getUsers: {
+      all: ["head-trainer"],
+      read: ["trainee", "trainer"],
+      write: ["trainer"],
+      delete: []
+    }
+  };
+  
+  function hasPermission(getUsers, role, permissionType) {
+    let {
+      getUsers: {
+        all: [a],
+        read: [b, c],
+        write: [d],
+        delete: []
+      }
+    } = permissions;
+    if (role == a) {
+      if (
+        permissionType == "all" ||
+        permissionType == "write" ||
+        permissionType == "read" ||
+        permissionType == "delete"
+      ) {
+        return console.log("True");
+      } else {
+        return console.log("False");
+      }
+    }
+    if (role == b) {
+      if (permissionType == "read") {
+        return console.log("True");
+      } else {
+        return console.log("False");
+      }
+    }
+    if (role == c) {
+      if (permissionType == "read" || permissionType == "write") {
+        return console.log("True");
+      } else {
+        return console.log("False");
+      }
+    }
+  }
+  
+  hasPermission("getUsers", "trainee", "write");
+  hasPermission("getUsers", "trainee", "getUsers");
```

Comment:
we should make this function like

If we call like hasPermission('getUsers', 'trainer', 'delete')

console.log(Trainer have permission to delete for the module getUsers)